### ISSUE TITLE
Other F# default Seq functions

### ISSUE THREAD

**Issue by lucasteles** [cognitedata/Expression#82]
The BCL F# Seq module has some other really helpful functions, do you have plans to implement them? need help/accept PR?

```fsharp
average
averageBy
cache
countBy
distinct
distinctBy
exactlyOne
exists
exists2
forall
forall2
groupBy
iteri
last
maxBy
minBy
nth
ofArray
ofList
pairwise
pick
readonly
reduce
skipWhile
sort
sortBy
sumBy
takeWhile
toArray
toList
truncate
tryFind
tryFindIndex
tryPick
where
windowed
zip3

```

**Comment by dbrattli** [comment]
Yes, would love a PR on this. Note that the code should be using `curry` or `curry_flipped` style functions instead of the nested (function returning) functions. See here for an example: https://github.com/cognitedata/Expression/blob/main/expression/collections/block.py#L669

**Comment by ShalokShalom** [comment]
Note: ```curry_flipped``` got renamed into ```curry_flip```

https://github.com/cognitedata/Expression/pull/87

**Comment by dbrattli** [comment]
Triage update: retaining the existing `PR welcome` status. The requested functions are independent additions; focused, separately tested pull requests are welcome, using the current `curry_flip` convention.

**Comment by bhargavandhe** [comment]
I’d like to contribute Seq.pairwise as a focused PR, including fluent and functional APIs, typing, laziness tests, and documentation. I couldn’t find an existing PR covering it.
